pilot/pkg/kube/inject/inject.go

```diff
@@ -43,17 +43,91 @@ import (
 )
 
 // per-sidecar policy and status
-const (
-	sidecarAnnotationPolicyKey                        = "sidecar.istio.io/inject"
-	sidecarAnnotationStatusKey                        = "sidecar.istio.io/status"
-	sidecarAnnotationProxyImageOverride               = "sidecar.istio.io/proxyImage"
-	sidecarAnnotationInterceptionModeKey              = "sidecar.istio.io/interceptionMode"
-	sidecarAnnotationIncludeOutboundIPRangesPolicyKey = "traffic.sidecar.istio.io/includeOutboundIPRanges"
-	sidecarAnnotationExcludeOutboundIPRangesPolicyKey = "traffic.sidecar.istio.io/excludeOutboundIPRanges"
-	sidecarAnnotationIncludeInboundPortsPolicyKey     = "traffic.sidecar.istio.io/includeInboundPorts"
-	sidecarAnnotationExcludeInboundPortsPolicyKey     = "traffic.sidecar.istio.io/excludeInboundPorts"
+var (
+	alwaysValidFunc = func(value string) error {
+		return nil
+	}
+
+	annotationValidators = make([]*annotationValidator, 0)
+
+	annotationPolicy = registerAnnotation(&annotationValidator{
+		name:      "sidecar.istio.io/inject",
+		validator: alwaysValidFunc,
+	})
+	annotationStatus = registerAnnotation(&annotationValidator{
+		name:      "sidecar.istio.io/status",
+		validator: alwaysValidFunc,
+	})
+	_ = registerAnnotation(&annotationValidator{
+		name:      "sidecar.istio.io/proxyImage",
+		validator: alwaysValidFunc,
+	})
+	_ = registerAnnotation(&annotationValidator{
+		name:      "sidecar.istio.io/interceptionMode",
+		validator: validateInterceptionMode,
+	})
+	_ = registerAnnotation(&annotationValidator{
+		name:      "traffic.sidecar.istio.io/includeOutboundIPRanges",
+		validator: ValidateIncludeIPRanges,
+	})
+	_ = registerAnnotation(&annotationValidator{
+		name:      "traffic.sidecar.istio.io/excludeOutboundIPRanges",
+		validator: ValidateExcludeIPRanges,
+	})
+	_ = registerAnnotation(&annotationValidator{
+		name:      "traffic.sidecar.istio.io/includeInboundPorts",
+		validator: ValidateIncludeInboundPorts,
+	})
+	_ = registerAnnotation(&annotationValidator{
+		name:      "traffic.sidecar.istio.io/excludeInboundPorts",
+		validator: ValidateExcludeInboundPorts,
+	})
 )
 
+func registerAnnotation(v *annotationValidator) *annotationValidator {
+	annotationValidators = append(annotationValidators, v)
+	return v
+}
+
+func validateAnnotations(annotations map[string]string) error {
+	for _, validator := range annotationValidators {
+		if err := validator.validate(annotations); err != nil {
+			return err
```

Comment:
consider returning a slice of error or multi-error or some kind. 
for validation we probably want to see all validation errors at once rather than aborting after first failure.